### PR TITLE
Adds EML processor

### PR DESCRIPTION
**Pull request by Oldharlem**
Adds a processor for .eml files
uses the `mail` gem for parsing the eml files.
Supports plain text in either single or multipart.

### CODE REVIEW

File: lib/langchain/processors/eml.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      def initialize(*)
+        depends_on 'mail'
+      end
+
+      # Parse the document and return the cleaned text
+      # @param [File] data
+      # @return [String]
+      def parse(data)
+        mail         = Mail.read(data.path)
+        text_content = extract_text_content(mail)
```
Comment (andreibondarev) [inline lib/langchain/processors/eml.rb:19]: I've used this gem before, have you seen it? https://github.com/github/email_reply_parser

**Comment by Oldharlem** [comment]
@andreibondarev anything else needed for the PR to be considered?

**Comment by andreibondarev** [comment]
@Oldharlem Thank you for the PR, I'd love to learn what you're using the gem for!

**Comment by Oldharlem** [comment]
> @Oldharlem Thank you for the PR, I'd love to learn what you're using the gem for!

My pleasure!
Langchain is part of our document processing process. We are building advocato.ai, a tool for legal professionals. 
The additional eml processor now allows them to upload their mail chains in EML format too to be chunked (by langchain) and used in a RAG powered research chat.

**Pull request merged**